core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala

```diff
@@ -80,6 +77,65 @@ private[spark] class DiskBlockManager(blockManager: BlockManager, conf: SparkCon
     new File(subDir, filename)
   }
 
+  def getFileDefault(filename: String): File = {
+    _getFile(filename, localDirs)
+  }
+  var getFile = getFileDefault _
+
+  private val hierarchyStore = conf.getOption("spark.storage.hierarchyStore")
+  if (hierarchyStore.isDefined) {
+    val HSLayers: Array[(String, Long)] =
+      hierarchyStore.get.trim.split(",").map {
+        s => val x = s.trim.split(" +")
+          (x(0), Utils.byteStringAsGb(x(1)))
```

Comment:
This number is the *threshold* before dropping into the next layer. 
We do get free space by using getFreeSpace method, like in getFileHierarchy.